### PR TITLE
New Package: Navi

### PR DESCRIPTION
**Pull request by delucca**
I've accidently closed #32570

Continuing here

@hyviquel

### CODE REVIEW

File: var/spack/repos/builtin/packages/z/package.py
Comment (iarspider) [inline var/spack/repos/builtin/packages/z/package.py]: Please remove the bolierplate (lines 6-21) and cleanup FIXME comments

**Review by iarspider — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/z/package.py
Comment (delucca) [reply var/spack/repos/builtin/packages/z/package.py]: Thanks, I've fixed it. That folder was supposed to be deleted

**Review by iarspider — DISMISSED** [review]
sha256 confirmed

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256, rest LGTM.

@denisidoro Are you willing to be the maintainer for this PR?  It means you'll notified of changes to this package to be given a chance to review them before they are merged.

File: var/spack/repos/builtin/packages/navi/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/navi/package.py:21]: Looks like you need a dependency on `cargo` added above (below the `version` directive).

File: var/spack/repos/builtin/packages/navi/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/navi/package.py]: Since there isn't a makefile associated with the package, I think you want to inherit from `Package` here.

```suggestion
class Navi(Package):
```

File: var/spack/repos/builtin/packages/navi/package.py
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/navi/package.py:21]: But that also means you need to add a package for installing it as well.

**Review by tldahlgren — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/navi/package.py
Comment (delucca) [reply var/spack/repos/builtin/packages/navi/package.py:21]: thanks, I've added the dependency to the file

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/navi/package.py:21]: Not everyone who may want to build this package will have `cargo` installed and no-one has added the package for it.

Adding the dependency is the first step but there still needs to be a package added for `cargo`.

**Review by tldahlgren — COMMENTED** [review]
Still need the `cargo` package.

File: var/spack/repos/builtin/packages/navi/package.py
Comment (delucca) [reply var/spack/repos/builtin/packages/navi/package.py:21]: But I've added the dependency on https://github.com/spack/spack/pull/32895/commits/299e3954bd8fcefdb13944881b81c2005c711b7e 🤔

Comment (delucca) [reply var/spack/repos/builtin/packages/navi/package.py:21]: sorry, I forgot that `cargo` is installed with `rust`.

Just changed it

**Review by tldahlgren — COMMENTED** [review]
I've tried to build several versions but am unable to install `rust` on my system to ensure `cargo` is included and available.

**Comment by tldahlgren** [comment]
We require accounts listed in maintainers to agree so we'll have to wait on @denisiidoro to respond.  Alternatively 😁 , you could replace his with your GitHub account and you'll be notified of any issues or changes to this package.

**Comment by delucca** [comment]
@tldahlgren I've replaced him with myself as maintainer :)

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.

**Comment by delucca** [comment]
@hyviquel

**Pull request merged**
